### ISSUE TITLE
Mullvad 2023.1 fails to work

### ISSUE THREAD

**Issue by HotCakeX** [mullvad/mullvadvpn-app#4376]
## Issue report

System information:
mullvad-product-version: 2023.1
os: Windows 11 Build 25300

### Issue description

After installing Mullvad version 2023.1, it's failing and doesn't work at all.

these are in the log

```
[2023-02-20 ][mullvad_daemon::version][INFO] Starting mullvad-daemon - 2023.1 2023-02-20
[2023-02-20 ][mullvad_daemon][INFO] Logging to C:\ProgramData\Mullvad VPN
[2023-02-20 ][mullvad_paths::settings][ERROR] Failed to obtain system app data path: Process ID for C:\Windows\system32\lsass.exe not found
[2023-02-20 ][mullvad_daemon][ERROR] Error: Unable to get settings dir
Caused by: Not able to find requested directory
```

and I found these in Event Viewer, Exploit Protection section:

```
Process '\Device\HarddiskVolume4\Program Files\Mullvad VPN\Mullvad VPN.exe' (PID 9348) was blocked from making system calls to Win32k.sys.

Process '\Device\HarddiskVolume4\Program Files\Mullvad VPN\Mullvad VPN.exe' (PID 8384) was blocked from making system calls to Win32k.sys.

Process '\Device\HarddiskVolume4\Program Files\Mullvad VPN\Mullvad VPN.exe' (PID 11348) was blocked from making system calls to Win32k.sys.

```

These are what I get when try to use the command line:

```
PS C:\Users\Admin> mullvad status
Error: Management interface error
Caused by: Management RPC server or client error
Caused by: transport error
Caused by: error trying to connect: The system cannot find the file specified. (os error 2)
Caused by: The system cannot find the file specified. (os error 2)
PS C:\Users\Admin>
```

**Comment by faern** [comment]
Thanks for reporting! We are looking into this issue.

**Comment by dlon** [comment]
For anyone who is having this problem: `mullvad-setup reset-firewall` will unblock your internet if the daemon won't start (though you will of course not be connected to the VPN).

**Comment by Occupant525** [comment]
If I may add to this (sorry if this is not protocol, I am new to Github), I can report that this isn't working on Linux either. I use ZorinOS. I have already reported it via the app itself but had to turn off the VPN. Please look into this too! Thanks.

**Comment by faern** [comment]
@Occupant525 That is definitely a completely different issue/bug. As this bug is about the Windows LSA system preventing us from doing certain syscalls. Linux does not run this code. The symptoms might be similar, but the issue is definitely another one.

We will not be able to help you with the little information you provided here. I hope the report you sent in the app will reach the app developers soon so we can look at it. But please mind that ZorinOS is not an officially supported distribution.

**Comment by Occupant525** [comment]
Hi, 

I see. Thanks for the info. 

What are the officially supported distros? 

The issue I sent via the app should definitely have been received and I also gave my Tutanota email with it so they can reach out if need be. 

Is there anything else I should know?

**Comment by Naxterra** [comment]
I have the same issue. It happens on 2023 versions including betas and latest public version. Beta version can't install Mullvad VPN Service, and public version is not able to start Mullvad VPN Service.

```
Log: \\?\C:\ProgramData\Mullvad VPN\daemon.log
====================
[2023-02-26 22:03:53.424][mullvad_daemon::version][INFO] Starting mullvad-daemon - 2023.1 2023-02-20
[2023-02-26 22:03:53.424][mullvad_daemon][INFO] Logging to C:\ProgramData\Mullvad VPN
[2023-02-26 22:03:53.425][mullvad_daemon::system_service][INFO] Service started.
[2023-02-26 22:03:53.425][mullvad_daemon::system_service][DEBUG] Update service status: StartPending, checkpoint: 1, wait_hint: 1s
[2023-02-26 22:03:53.455][mullvad_paths::settings][ERROR] Failed to obtain system app data path: Process ID for C:\WINDOWS\system32\lsass.exe not found
[2023-02-26 22:03:53.455][mullvad_daemon::system_service][ERROR] Error: Unable to get settings dir
Caused by: Not able to find requested directory
[2023-02-26 22:03:53.455][mullvad_daemon::system_service][DEBUG] Update service status: Stopped, checkpoint: 0, wait_hint: 0ns
[2023-02-26 22:03:53.455][mullvad_daemon][DEBUG] Process exiting with code 0

====================
Log: \\?\C:\ProgramData\Mullvad VPN\install.log
====================
[2023-02-26 22:02:32.421] Running installer for Mullvad VPN 2023.1.0
[2023-02-26 22:02:32.422] Windows version: Windows 11 Enterprise, 10.0.22621.1265
[2023-02-26 22:02:32.429] MigrateCache()
[2023-02-26 22:02:32.447] Failed to migrate cache: Could not find named process (process.cpp: 57)
[2023-02-26 22:02:32.447] RemoveRelayCache()
[2023-02-26 22:02:32.449] RemoveRelayCache() completed successfully
[2023-02-26 22:02:32.449] RemoveApiAddressCache()
[2023-02-26 22:02:32.450] RemoveApiAddressCache() completed successfully
[2023-02-26 22:02:32.454] RemoveAbandonedWintunAdapter()
[2023-02-26 22:02:32.567] RemoveAbandonedWintunAdapter() completed successfully
[2023-02-26 22:02:32.567] RemoveSplitTunnelDriver()
[2023-02-26 22:02:32.568] Removing Split Tunneling driver
[2023-02-26 22:02:32.677] RemoveSplitTunnelDriver() completed successfully
[2023-02-26 22:02:32.677] InstallService()
[2023-02-26 22:02:32.678] Running "mullvad-daemon" for it to self-register as a service
[2023-02-26 22:02:32.964] Starting service
[2023-02-26 22:02:33.086] InstallService() completed successfully
[2023-02-26 22:02:33.089] AddCLIToEnvironPath()
[2023-02-26 22:02:33.590] AddCLIToEnvironPath() completed successfully
[2023-02-26 22:02:33.590] InstallTrayIcon()
[2023-02-26 22:02:33.595] InstallTrayIcon() completed successfully

====================
Log: \\?\~\AppData\Local\Mullvad VPN\logs\frontend-main.log
====================
[2023-02-26 22:03:44.114][verbose] Chromium sandbox is enabled
[2023-02-26 22:03:44.115][info] Running version 2023.1
[2023-02-26 22:03:44.144][info] Detected locale: de
[2023-02-26 22:03:44.154][info] Loaded translations de/messages
[2023-02-26 22:03:44.155][info] Loaded translations de/relay-locations
[2023-02-26 22:03:54.160][error] Failed to connect to daemon: Failed to connect before the deadline
[2023-02-26 22:03:54.161][error] Unable to connect to daemon: Failed to connect before the deadline
[2023-02-26 22:04:07.187][error] Failed to connect to daemon: Failed to connect before the deadline
[2023-02-26 22:04:07.187][error] Failed to reconnect - Error: Failed to connect before the deadline
[2023-02-26 22:04:20.189][error] Failed to connect to daemon: Failed to connect before the deadline
[2023-02-26 22:04:20.189][error] Failed to reconnect - Error: Failed to connect before the deadline
[2023-02-26 22:04:33.201][error] Failed to connect to daemon: Failed to connect before the deadline
[2023-02-26 22:04:33.201][error] Failed to reconnect - Error: Failed to connect before the deadline
[2023-02-26 22:04:46.217][error] Failed to connect to daemon: Failed to connect before the deadline
[2023-02-26 22:04:46.217][error] Failed to reconnect - Error: Failed to connect before the deadline
[2023-02-26 22:04:59.223][error] Failed to connect to daemon: Failed to connect before the deadline
[2023-02-26 22:04:59.224][error] Failed to reconnect - Error: Failed to connect before the deadline
[2023-02-26 22:05:12.236][error] Failed to connect to daemon: Failed to connect before the deadline
[2023-02-26 22:05:12.236][error] Failed to reconnect - Error: Failed to connect before the deadline
[2023-02-26 22:05:25.266][error] Failed to connect to daemon: Failed to connect before the deadline
[2023-02-26 22:05:25.266][error] Failed to reconnect - Error: Failed to connect before the deadline
[2023-02-26 22:05:38.289][error] Failed to connect to daemon: Failed to connect before the deadline
[2023-02-26 22:05:38.289][error] Failed to reconnect - Error: Failed to connect before the deadline

====================
Log: \\?\~\AppData\Local\Mullvad VPN\logs\frontend-renderer.log
====================
[2023-02-26 22:03:44.497][info] Loaded translations de/messages
[2023-02-26 22:03:44.497][info] Loaded translations de/relay-locations
[2023-02-26 22:03:44.497][verbose] Tunnel state: disconnected
[2023-02-26 22:03:44.501][error] Failed to update the location: No connection established to daemon
[2023-02-26 22:03:44.501][error] Failed to update the location: No connection established to daemon
```

**Comment by HotCakeX** [comment]
>  Beta version can't install Mullvad VPN Service

I've had the same problem too. Curious to know how the beta got promoted to release version while having such fundamental important problems.

**Comment by faern** [comment]
We are working on a new release with this fixed on Windows. Sadly none of our internal testers nor beta program users had LSA enabled in their Windows installs it seems.

**Comment by Naxterra** [comment]
> We are working on a new release with this fixed on Windows. Sadly none of our internal testers nor beta program users had LSA enabled in their Windows installs it seems.

Are they using older version of Windows or without any security measurements active? It is enabled by default in Windows 11 22H2
Ref: https://learn.microsoft.com/en-us/windows-server/security/credentials-protection-and-management/configuring-additional-lsa-protection

If you need more logs or need a volunteer for testing, I can help.

**Comment by faern** [comment]
> Are they using older version of Windows or without any security measurements active? It is enabled by default in Windows 11 22H2

We are testing on Windows 11 but it might be that we had not yet upgraded to 22H2 or it might be that it stays disabled if you only upgrade win11 from an older version? I'm not fully sure. We don't make clean installs of all operating systems from scratch for every testing round. However, we are now, in parallel with this, working on adding automated installation tests with LSA enabled, to catch stuff like this in the future.

> If you need more logs or need a volunteer for testing, I can help.

Thank you, but no need. We have already merged the fix (#4386), and are working on building a release.

**Comment by faern** [comment]
Yeah, upgrading an existing win11 install to 22H2 does not enable LSA. So this is only a problem for people with a fresh install of that version.

**Comment by HotCakeX** [comment]
If i may chime in, I have a few reasonable suggestions that I think will help Mullvad going forward:

1. If it isn't the case already, make Mullvad app send logs, errors and crash reports automatically to the devs, without the need for user intervention. I encountered this bug early on but didn't report it because at that time, I was too busy with other matters so I just removed the beta and installed stable version quickly to continue my work and had hoped it would be fixed when it makes the released version.
2. It wouldn't obviously make sense to clean install every operation system in the world where Mullvad works on to test it, but it does make sense to do it for one or two OSes at least, one of them being Windows. Grabbing a fresh ISO file from MSFT website and setting up a Hyper-V VM takes only a few minutes. Plus, Mullvad app is updated only a couple of times a year. With a VM checkpoint, it's super easy to manage and test as many times as you want.
3. Check [Windows insider blog](https://blogs.windows.com/blog/tag/windows-insider-program/) regularly for release notes and stay informed about the upcoming features, specially security features. It'd also be helpful to use insider builds in release preview channel instead, to see the upcoming features making their way to the stable release beforehand and be ready for them.


I'm glad a new version is on its way, thank you!

**Comment by Naxterra** [comment]
For the internal employees, sending logs automatically will make sense but you can't do it on customer devices without approval.

Virtual machines would be good idea. They should have their own test environments. Well, that is what testing is for, right? They should test on every officially supported OS versions. Because a simple overlook like this can affect thousands of customers.

Insider blogs are helpful, but still they are in development phase and can be changed very frequently. That will make it very hard to follow.

**Comment by HotCakeX** [comment]
> For the internal employees, sending logs automatically will make sense but you can't do it on customer devices without approval.
> 
> Virtual machines would be good idea. They should have their own test environments. Well, that is what testing is for, right? They should test on every officially supported OS versions. Because a simple overlook like this can affect thousands of customers.
> 
> Insider blogs are helpful, but still they are in development phase and can be changed very frequently. That will make it very hard to follow.

I'd assume there'd be something like a checkbox in Mullvad settings, enabled by default with the option to disable it if someone wants to. Telemetry and error reporting is necessary because not everyone bothers to create actionable proper feedback. It's not like the logs are gonne include browsing history.

Windows insider Dev builds do change frequently and they aren't tied to any specific release. Beta channel build changes aren't frequent and mostly contain code that will come to next build of Windows. I suggested release preview channel, the least dynamic channel, I don't think there's anything hard about it, it's the most stable channel.

Even insider Dev builds are very solid and stable, that's what I've using for the past few years nonstop, because the other channels were too stable and boring lol

**Comment by Naxterra** [comment]
For now only option is to send logs manually. Telemetry and error reporting are necessary but shouldn't be forced. If it is optional, then it is good, but if it sends logs automatically, then you will have lots of complaints, as well as problems with the privacy laws. Everything must be documented, and checked against compliance rules too.

**Comment by HotCakeX** [comment]
@faern Understandable, thank you!

btw, has there been any progress towards extending your digital signatures to other DLLs and executables in Mullvad app? if not, when do you think that can happen? thanks

**Comment by HJLBX** [comment]
> > Are they using older version of Windows or without any security measurements active? It is enabled by default in Windows 11 22H2
> 
> We are testing on Windows 11 but it might be that we had not yet upgraded to 22H2

What? Are you not doing regular testing after Windows updates in a virtual machine? Do you have a dedicated internal QA team?

> or it might be that it [LSA] stays disabled if you only upgrade win11 from an older version? 

No

> We don't make clean installs of all operating systems from scratch for every testing round. 

You do not need to do clean installs of the OS for every test round. You just do manual QA testing in a clean virtual machine running Windows Insider builds and manage the snapshots correctly to keep every updated box clean. I do QA for a living and if all your testing is automated, then it is clear why there are these nagging ongoing QA\QC problems on Windows 11 22H2. Automated testing is mainly for regression testing. We are a small team of 4 and we manage to do extensive manual QA\QC after every single Windows update - every month! We get full coverage across workstations and server editions!

Anyway, Windows 11 22H2 was released on 20/9/2022. Seems like you have some real QA\QC issues.

**Comment by faern** [comment]
> You do not need to do clean installs of the OS for every test round. You just do manual QA testing in a clean virtual machine running Windows Insider builds and manage the snapshots correctly to keep every updated box clean. I do QA for a living and if all your testing is automated, then it is clear why there are these nagging ongoing QA\QC problems on Windows 11 22H2

It's the inverse. We have historically, and are still doing, 95% of testing fully manually before each release. It's only the last few months that we also get an automated testing framework up and running continuously.

Yes, we apparently do need a clean install. Because *upgrades* of Windows 11 does not enable LSA (the main culpit here). So the only way to trigger the issue here would have been a fresh install of the latest Windows 11 version.

We run through a manual test sequence that takes about an hour per platform, and we do this on 11 platforms. So it's a huge time sink.

**Comment by faern** [comment]
This has now been fixed and we finally cut a release. Please upgrade to `2023.2` to have the LSA issues fixed! https://github.com/mullvad/mullvadvpn-app/releases/tag/2023.2

The release is not yet, but will very shortly be on our website.

**Comment by faern** [comment]
Awesome! Thanks for the quick feedback :heart:

**Comment by citrixscu** [comment]
Also confirmed working on Windows 11 22H2 Dev (OS Build 25309.1000) in a VM using 22H2 host under VMware Workstation Pro 17.x.

**Comment by yslupdates** [comment]
Hi I've encountered this issue on 2023.3 for debian 11 (this is a LXC within proxmox)

Installed using these instructions https://mullvad.net/en/help/install-mullvad-app-linux/
```

root@mullvadvpn:~# mullvad status
Error: Management interface error
Caused by: Management RPC server or client error
Caused by: transport error
Caused by: error trying to connect: No such file or directory (os error 2)
Caused by: No such file or directory (os error 2)
root@mullvadvpn:~# mullvad version
Error: Management interface error
Caused by: Management RPC server or client error
Caused by: transport error
Caused by: error trying to connect: No such file or directory (os error 2)
Caused by: No such file or directory (os error 2)
```

**Comment by faern** [comment]
@yslupdates This issue is about a Windows specific issue we had that was fixed. Please file a new issue if you can't find any that fits your issue. 

It looks like your `mullvad-daemon` is not running. You can check the status of the service with `systemctl status mullvad-daemon.service`. If you open a new issue, please include logs from the daemon.
